### PR TITLE
[CSL-1963] Remove v0-prefixed version of the API

### PR DESCRIPTION
**Pull request by adinapoli-iohk**
This PR removes the v0-prefixed version of the API from `wallet-new`, to make things less confusing.

Swagger docs updated to reflect that.

### CODE REVIEW

File: wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs [hunk 1]
```diff
@@ -383,17 +383,18 @@ of [Cardano](https://github.com/input-output-hk/cardano-sl).
 ### Request format (all versions)
 
 Issuing requests against this API is conceptually not very different from any other web service out there. The API
-is versioned, meaning is possible to access different versions of the latter by changing the _version number_ in the URL.
+is **versioned**, meaning is possible to access different versions of the latter by changing the _version number_ in the URL.
```
Comment (ksaric) [inline wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs]: "meaning it's possible"
"latter by changing" -> "new api by adding the"

File: wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs [hunk 2]
```diff
@@ -383,17 +383,18 @@ of [Cardano](https://github.com/input-output-hk/cardano-sl).
...
 
-For example, _omitting_ the version number of passing `v0` would call the version 0 of the API. Examples:
+**For the sake of backward compatibility, we expose the legacy version of the API, available simply as
+unversioned endpoints.**
+
+This means that _omitting_ the version number would call the version 0 of the API. Examples:
```
Comment (ksaric) [inline wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs]: "version 0" -> "old version"?

File: wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs (hunk 1 above)
Comment (adinapoli-iohk) [reply wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs]: good catch!

File: wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs (hunk 2 above)
Comment (adinapoli-iohk) [reply wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs]: Yea, I like it more 😉

**Comment by adinapoli-iohk** [comment]
Superseded by https://github.com/input-output-hk/cardano-sl/pull/2026

**Pull request closed without merging**